public/app/core/services/NewFrontendAssetsChecker.ts

```diff
@@ -0,0 +1,51 @@
+import { getBackendSrv, getGrafanaLiveSrv } from '@grafana/runtime';
+
+export class NewFrontendAssetsChecker {
+  private hasUpdates = false;
+  private previous = '';
+  private interval = 1000 * 60 * 4; // force check after 5 mins
+  private checked = Date.now() - this.interval / 5;
+
+  public start() {
+    const startup = Date.now();
+    let prevState = false;
+
+    // Subscribe to live connection state changes and check for new assets when re-connected
+    const live = getGrafanaLiveSrv();
+    if (live) {
+      live.getConnectionState().subscribe((connected) => {
+        const elapsed = Date.now() - startup;
+        if (elapsed > 1000 && connected && connected !== prevState) {
+          this._checkForUpdates();
+        }
+        prevState = connected;
+      });
+    }
+  }
+
+  private async _checkForUpdates() {
+    if (this.hasUpdates) {
+      return;
+    }
+
+    const resultRaw = await getBackendSrv().get('/api/frontend/assets');
+    const result = JSON.stringify(resultRaw);
+    if (this.previous?.length && this.previous !== result) {
+      this.hasUpdates = true;
+    }
+    this.previous = result;
+    this.checked = Date.now();
+  }
+
+  /** This is called on page navigation events */
+  public reloadIfUpdateDetected() {
+    if (this.hasUpdates) {
```

Comment:
i was about to make this comment. 👍 

checking for minimum of a few mins since last keyboard or mouse activity should mitigate much of this. also maybe PageVisibility api.

Reply:
@academo this is only called when moving to another page

Reply:
> navigation events trigger even in unsaved work.
@academo do you have an example of unsaved work between two different routes? that should have a Prompt in that case, which will block navigation. This is what dashboards do at least.

Reply:
but technically yes, an edit UI could be broken up into different routes and be missing a Prompt and beforeunload handler to block navigation, in that case we could lose unsaved work. But do not think we have any such page in Grafana core, and for plugins there is only one route so there is no unmount of GrafanaRoute between same plugin page navigations. 

We could limit it to page changes between sections and dashboards (where we know there would be no unsaved work) to be 100% sure.

Reply:
@torkelo IMO it is better to be cautious in this case and only trigger this in navigation events we consider safe.

Reply:
There's also that case of people who put grafana on a big screen and forget about it for days at a time. 

It goes out of the scope of this initial PR but we can enhance this functionality by checking if you are in a dashboard page, haven't used any interaction in a long while (hours) and the dashboard has no unsaved work.